File: app/src/main/java/com/sortedqueue/programmercreek/adapter/SubTestPagerAdapter.java
package com.sortedqueue.programmercreek.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.sortedqueue.programmercreek.database.ProgramTable;
import com.sortedqueue.programmercreek.fragments.CodeViewFragment;
import com.sortedqueue.programmercreek.fragments.SubTestFragment;

import java.util.ArrayList;

/**
 * Created by Alok Omkar on 2017-01-26.
 */

public class SubTestPagerAdapter extends FragmentPagerAdapter {

    private CodeViewFragment codeViewFragment;
    private ArrayList<ProgramTable>[] programTableArray;
    private ArrayList<Fragment> subTestFragments;

    public SubTestPagerAdapter(FragmentManager fm, ArrayList<ProgramTable>[] programTableArray) {
        super(fm);
        this.programTableArray = programTableArray;
        subTestFragments = new ArrayList<>();
        int index = 0;
        for( ArrayList<ProgramTable> programTables : programTableArray ) {
            SubTestFragment subTestFragment = new SubTestFragment();
            subTestFragment.setProgramTables( programTables );
            subTestFragment.setIndex(index++);
            subTestFragments.add(subTestFragment);
        }
        codeViewFragment = new CodeViewFragment();
        subTestFragments.add(codeViewFragment);

    }

    public CodeViewFragment getCodeViewFragment() {
        return codeViewFragment;
    }

    @Override
    public Fragment getItem(int position) {
        return subTestFragments.get(position);
    }

    @Override
    public int getCount() {
        return programTableArray.length + 1;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if( position == programTableArray.length ) {
            return "Output";
        }
        ArrayList<ProgramTable> programTables = programTableArray[position];

        String pageTitle = programTables.get(0).getLine_No() + " - " + programTables.get(programTables.size() - 1).getLine_No();
        return pageTitle;
    }

}

File: app/src/main/java/com/sortedqueue/programmercreek/adapter/SubTestPagerAdapter.kt
package com.sortedqueue.programmercreek.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import com.sortedqueue.programmercreek.database.ProgramTable
import com.sortedqueue.programmercreek.fragments.CodeViewFragment
import com.sortedqueue.programmercreek.fragments.SubTestFragment
import com.sortedqueue.programmercreek.interfaces.SubTestCommunicationListener

import java.util.ArrayList

/**
 * Created by Alok Omkar on 2017-01-26.
 */

class SubTestPagerAdapter(fm: FragmentManager, private val programTableArray: Array<ArrayList<ProgramTable>>, subTestCommunicationListener: SubTestCommunicationListener) : FragmentPagerAdapter(fm) {

    val codeViewFragment: CodeViewFragment
    private val subTestFragments: ArrayList<Fragment>

    init {
        subTestFragments = ArrayList<Fragment>()
        var index = 0
        for (programTables in programTableArray) {
            val subTestFragment = SubTestFragment()
            subTestFragment.setProgramTables(programTables)
            subTestFragment.setIndex(index++)
            subTestFragment.setSubmitTestCommunicationListener(subTestCommunicationListener)
            subTestFragments.add(subTestFragment)
        }
        codeViewFragment = CodeViewFragment()
        codeViewFragment.setTotalModules(programTableArray.size)
        subTestFragments.add(codeViewFragment)

    }

    override fun getItem(position: Int): Fragment {
        return subTestFragments[position]
    }

    override fun getCount(): Int {
        return programTableArray.size + 1
    }

    override fun getPageTitle(position: Int): CharSequence {
        if (position == programTableArray.size) {
            return "Preview"
        }
        if (programTableArray.size == 1) {
            return "Test"
        }
        val programTables = programTableArray[position]

        val pageTitle = programTables[0].line_No.toString() + " - " + programTables[programTables.size - 1].line_No
        return pageTitle
    }

}
